.pre-commit-config.yaml

```diff
@@ -1,12 +1,24 @@
 repos:
+  - repo: https://github.com/pycqa/isort
+    rev: 5.12.0
+    hooks:
+      - id: isort
+        language_version: python3
+
   - repo: https://github.com/psf/black
     rev: 22.10.0
     hooks:
-    - id: black
-      language_version: python3
+      - id: black
+        language_version: python3
 
   - repo: https://github.com/pycqa/flake8
     rev: 5.0.4
     hooks:
-    - id: flake8
-      language_version: python3
+      - id: flake8
+        language_version: python3
+
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.2
+    hooks:
+      - id: codespell
+        language_version: python3
```

Comment:
I've never used `codespell` before, but it seems nice. It did find a bunch of false-positives in the code files, for now I've excluded all code files so it just runs on our docs.

Note that when adding new pre-commit hooks you'll need to run them manually on every file to ensure things pass. I did this using `pre-commit run --all-files` and fixed the issues that popped up.